### PR TITLE
Make TagIteratorPool.Close idempotent

### PR DESCRIPTION
**Pull request by ryanhall07**
Previously a double call to Close would result in the tag iterator being
added back to the pool twice, which would then result in the same
iterator being given back out to multiple callers later.

Additionally removed the unnecessary call to tagIter.Close() in
Cluster.Query, since closing the results will close the tag iterator.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/dbnode/network/server/tchannelthrift/cluster/service.go [hunk 1]
```diff
@@ -140,6 +140,8 @@ func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryR
 			return nil, convert.ToRPCError(err)
 		}
 
+		defer results.Finalize()
```
Comment (ryanhall07) [inline src/dbnode/network/server/tchannelthrift/cluster/service.go:143]: we were also leaking resources on errors :(

File: src/x/ident/identifier_pool_test.go [hunk 2]
```diff
@@ -210,11 +211,46 @@ func (s idPoolTestSuite) TestPoolTagsIterator() {
...
+	i1 := pool.TagsIterator()
+	i2 := pool.TagsIterator()
+	i1.Reset(tags)
+	i2.Reset(tags)
+	require.Equal(t, 2, i1.Remaining())
+	require.Equal(t, 2, i2.Remaining())
+
+	// close one of the iterators and ensure it does not close the other.
+	i1.Close()
+	require.Equal(t, 2, i2.Remaining())
```
Comment (ryanhall07) [inline src/x/ident/identifier_pool_test.go:242]: this test failed with the bug fix

File: src/dbnode/network/server/tchannelthrift/cluster/service.go (hunk 1 above)
Comment (robskillington) [reply src/dbnode/network/server/tchannelthrift/cluster/service.go:143]: Ah good catch.

File: src/x/ident/tag_iterator.go [hunk 3]
```diff
@@ -139,6 +140,10 @@ func (i *tagSliceIter) Err() error {
 }
 
 func (i *tagSliceIter) Close() {
+	if i.closed {
+		return
+	}
+	i.closed = true
```
Comment (robskillington) [inline src/x/ident/tag_iterator.go:146]: Will need to set closed = false for both Reset and ResetFields I believe so will close properly after the two types of resets are used/called?

**Review by robskillington — APPROVED** [review]
LGTM but do think we should update the Reset and ResetFields methods on the tag iterator so that closed is set to false properly before reuse again

File: src/x/ident/tag_iterator.go (hunk 3 above)
Comment (ryanhall07) [reply src/x/ident/tag_iterator.go:146]: ugh. good catch. will update the test.

**Pull request merged**